src/components/files/createDefaultFolders.ts

```diff
@@ -0,0 +1,57 @@
+// Creates (and populates if appropriate) the folders that are not saved as part
+// of git repo. We run this for safety when starting the server, or before
+// saving or loading a snapshot.
+
+import {
+  PREFERENCES_FOLDER,
+  PREFERENCES_FILE_NAME,
+  LOCALISATION_FOLDER,
+  FILES_FOLDER,
+  SNAPSHOT_FOLDER,
+  BASE_SNAPSHOT_FOLDER,
+  GENERIC_THUMBNAILS_FOLDER,
+  GENERIC_THUMBNAILS_SOURCE_FOLDER,
+  DATABASE_FOLDER,
+  BASE_SNAPSHOT_NAME,
+} from '../../constants'
+import fs from 'fs'
+import path from 'path'
+import { execSync } from 'child_process'
+import { makeFolder } from '../utilityFunctions'
+
+export function createDefaultDataFolders() {
```

Comment:
This is the major "catch-all" function, which combines a lot of things that were done piecemeal elsewhere. It ensures that whenever the server starts or a snapshot is saved or loaded, there will always be a valid "files", "snapshots", "localisations" and "preferences" folders with (at the very least) contents from core_templates.